pkgs/tools/networking/ntpsec/default.nix

```
@@ -0,0 +1,73 @@
+{ stdenv, lib, fetchurl, 
+  wafHook, pkgconfig, gnum4,
+  asciidoc, libxslt, docbook_xsl,
+  bison, openssl_1_1, libbsd,
+  coreutils, pythonPackages, pps-tools,
+  libcap ? null, libseccomp ? null,
+}:
+
+assert stdenv.isLinux -> libcap != null;
+assert stdenv.isLinux -> libseccomp != null;
+
+let
+  python = pythonPackages.python;
+
+  withSeccomp = stdenv.isLinux && (stdenv.isi686 || stdenv.isx86_64);
+in
+
+stdenv.mkDerivation rec {
+  name = "ntpsec-${version}";
+  version = "1.1.4";
+
+  src = fetchurl {
+    url = "https://ftp.ntpsec.org/pub/releases/ntpsec-${version}.tar.gz";
+    sha256 = "1vv5ckjlmaa9p2ri3fnqh804fxn0wl31jkkjn6v6hfw5z69vkwbn";
+  };
+
+  patches = [
+    ./skip-decodenetnum-tests.patch
+  ];
+
+  preBuild = ''
+    substituteInPlace ./ntpclients/ntpleapfetch \
+      --replace cat ${coreutils}/bin/cat \
+      --replace basename ${coreutils}/bin/basename \
+      --replace chmod ${coreutils}/bin/chmod
+  '';
+
+  wafConfigureFlags = [
+    "--sysconfdir=/etc"
+    "--localstatedir=/var"
+    "--refclock=all"
+    "--enable-leap-smear"
+    "--python=${python}/bin/python"
+  ]
+    ++ stdenv.lib.optional withSeccomp "--enable-seccomp";
+
+
+  nativeBuildInputs = [ pkgconfig wafHook gnum4
+    asciidoc docbook_xsl libxslt
+    python pythonPackages.wrapPython
+  ];
+
+  buildInputs = [ libbsd libcap bison openssl_1_1 python ]
+    ++ lib.optional withSeccomp libseccomp
+    ++ lib.optional stdenv.isLinux pps-tools;
+
+  postFixup = ''
+    wrapPythonPrograms
+  '';
+
+  hardeningEnable = [ "pie" ];
+
+  meta = with stdenv.lib; {
+    homepage = https://www.ntpsec.org/;
+    description = "A secure, hardened, and improved implementation of Network Time Protocol derived from NTP Classic";
+    license = {
```

You can list many
```suggestion
    license = with licenses; [ bsd2 bsd3 mit ];
```
for completeness, you can add the ntp license, if you want.